### PR TITLE
[PT-5597] (2) Updating catalog `construct_search_parameters`

### PR DESCRIPTION
**Pull request by andher1802**
See changelog.

Checklist:
* [x] Test coverage
* [x] Version bump
* [x] Changelog update

### CODE REVIEW

File: up42/catalog.py [hunk 1]
```diff
@@ -177,7 +183,7 @@ def construct_search_parameters(
             elif usage_type == ["DATA", "ANALYTICS"]:
```
Comment (javidq) [inline up42/catalog.py:183]: Let's simplify this block to:
```python
            if set(usage_type) <= {"DATA", "ANALYTICS"}:
                query_filters["up42:usageType"] = {"in": usage_type}
            else:
                raise InvalidUsageType("usage_type is invalid")
```

File: tests/test_catalog.py [hunk 2]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+        response = {**SEARCH_PARAMETERS}
+        optional_response: dict = {"query": {}}
+        if "max_cloudcover" in max_cloudcover:
+            optional_response["query"]["cloudCoverage"] = {"lte": max_cloudcover["max_cloudcover"]}
+        if "usage_type" in usage_type:
+            optional_response["query"]["up42:usageType"] = {"in": usage_type["usage_type"]}
+        response = {**response, **optional_response}
+        assert self.catalog.construct_search_parameters(**params) == response
+
+    def test_should_fail_construct_search_parameters_with_wrong_data_usage(self):
```
Comment (javidq) [inline tests/test_catalog.py:278]: should_fail -> fails_to

File: tests/test_catalog.py [hunk 3]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+            "geometry": SIMPLE_BOX,
+            "collections": [PHR],
+            "start_date": START_DATE,
+            "end_date": END_DATE,
+            **usage_type,
+            **max_cloudcover,
+        }
+        response = {**SEARCH_PARAMETERS}
+        optional_response: dict = {"query": {}}
+        if "max_cloudcover" in max_cloudcover:
```
Comment (javidq) [inline tests/test_catalog.py:271]: if max_cloudcover

File: tests/test_catalog.py [hunk 4]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+            optional_response["query"]["cloudCoverage"] = {"lte": max_cloudcover["max_cloudcover"]}
+        if "usage_type" in usage_type:
```
Comment (javidq) [inline tests/test_catalog.py:273]: if usage_type

File: tests/test_catalog.py [hunk 5]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+    def test_should_construct_search_parameters(self, usage_type: dict, max_cloudcover: dict):
+        params = {
+            "geometry": SIMPLE_BOX,
+            "collections": [PHR],
+            "start_date": START_DATE,
+            "end_date": END_DATE,
+            **usage_type,
+            **max_cloudcover,
+        }
+        response = {**SEARCH_PARAMETERS}
```
Comment (javidq) [inline tests/test_catalog.py:269]: Not needed

File: tests/test_catalog.py [hunk 6]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+        optional_response: dict = {"query": {}}
+        if "max_cloudcover" in max_cloudcover:
+            optional_response["query"]["cloudCoverage"] = {"lte": max_cloudcover["max_cloudcover"]}
+        if "usage_type" in usage_type:
+            optional_response["query"]["up42:usageType"] = {"in": usage_type["usage_type"]}
+        response = {**response, **optional_response}
```
Comment (javidq) [inline tests/test_catalog.py:275]: `response = {**SEARCH_PARAMETERS, **optional_response}`

File: tests/test_catalog.py [hunk 7]
```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+        params = {
+            "geometry": SIMPLE_BOX,
+            "collections": [PHR],
+            "start_date": START_DATE,
+            "end_date": END_DATE,
+            **usage_type,
+            **max_cloudcover,
+        }
+        response = {**SEARCH_PARAMETERS}
+        optional_response: dict = {"query": {}}
```
Comment (javidq) [inline tests/test_catalog.py:270]: Why do we need type hint here?

File: up42/catalog.py [hunk 8]
```diff
@@ -177,7 +183,7 @@ def construct_search_parameters(
...
                 query_filters["up42:usageType"] = {"in": ["DATA", "ANALYTICS"]}
```
Comment (javidq) [inline up42/catalog.py:184]: Why do we need type ignore on L176?

File: tests/test_catalog.py [hunk 9]
```diff
@@ -277,45 +297,6 @@ def test_construct_order_parameters(catalog_mock):
...
-        ) == {
-            "datetime": "2014-01-01T00:00:00Z/2020-12-31T23:59:59Z",
-            "intersects": SIMPLE_BOX,
-            "limit": 1,
-            "collections": [PHR],
-            "query": {"up42:usageType": {"in": params["usage_type"]}},
-        }
-
-
 def test_estimate_order_from_catalog(requests_mock, auth_mock):
```
Comment (javidq) [inline tests/test_catalog.py:300]: test_should_estimate_order + type hints on params

File: tests/test_catalog.py [hunk 10]
```diff
@@ -15,17 +14,18 @@
...
     "collections": [PHR],
-    "limit": 4,
-    "query": {
-        "cloudCoverage": {"lte": 20},
-        "up42:usageType": {"in": ["DATA", "ANALYTICS"]},
-    },
+    "limit": 10,
 }
 
+Geometry = catalog.Geometry
```
Comment (javidq) [inline tests/test_catalog.py:27]: Not needed

**Review by javidq — CHANGES_REQUESTED** [review]

File: up42/catalog.py (hunk 8 above)
Comment (andher1802) [reply up42/catalog.py:184]: not needed, it came from before.

File: tests/test_catalog.py (hunk 7 above)
Comment (andher1802) [reply tests/test_catalog.py:270]: `mypy` complains about it. I believe the assignment of the query dictionary makes it required. 
Screenshot 2024-09-25 at 12 09 38

**Review by javidq — APPROVED** [review]

**Pull request merged**